common/utils/src/main/resources/error/error-classes.json

```diff
@@ -2024,6 +2024,11 @@
           "expects an integer value in [0, <upper>), but got <invalidValue>."
         ]
       },
+      "CHARSET" : {
+        "message" : [
+          "expects one of the charsets 'US-ASCII', 'ISO-8859-1', 'UTF-8', 'UTF-16BE', 'UTF-16LE', 'UTF-16', but got <charset>."
```

Comment:
Should we give advice on the legacy configuration?

Reply:
I plan to restrict the supported charsets in the code, and add a config for the legacy behaviour. In the following PR, I will modify the message with an advice.